core/src/main/scala/za/co/absa/fadb/status/aggregation/StatusAggregator.scala

```diff
@@ -0,0 +1,56 @@
+/*
+ * Copyright 2022 ABSA Group Limited
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package za.co.absa.fadb.status.aggregation
+
+import za.co.absa.fadb.exceptions.StatusException
+import za.co.absa.fadb.status.{ExceptionOrStatusWithDataResultAgg, ExceptionOrStatusWithDataRow, FunctionStatusWithData}
+
+/**
+  *  `StatusAggregator` is a base trait that defines the interface for aggregating the error statuses of a function
+  *  invocation. It provides methods to aggregate the error statuses into a single status information - this is
+  *  typically needed for database functions that retrieve multiple records.
+  */
+trait StatusAggregator {
+
+  private [aggregation] def gatherExceptions[R](
```

Comment:
We don't need to iterate over the whole sequence for certain aggregation approaches.

Reply:
Yeah in most algorithms we won't, but for some perhaps we'll need, so I kept it here for now